### DISCUSSION TITLE
Fresh Install on ARM64 Debian 12 Go Boom!

### DISCUSSION THREAD

**Discussion by hackdefendr** [joshuar/go-hass-agent#292] [Q&A]
Hi Lo everyone,

Am I missing something in my setup? I have no idea, but probably yes. I following the steps for installation and initializing the configuration. My logs only show ERR when connecting to any default sensor due to 502 BAD GATEWAY.

```
Sep 27 10:37:58 atlas go-hass-agent[3218707]: Sep 27 10:37:58.720 ERR Process sensor failed. error="failed to send sensor update request for cpufreq_core73_frequency: code 502: 502 Bad Gateway"
Sep 27 10:37:58 atlas go-hass-agent[3218707]: Sep 27 10:37:58.722 ERR Process sensor failed. error="failed to send sensor update request for core_79_cpu_usage: code 502: 502 Bad Gateway"
Sep 27 10:37:58 atlas go-hass-agent[3218707]: Sep 27 10:37:58.722 ERR Process sensor failed. error="failed to send sensor update request for cpufreq_core87_frequency: code 502: 502 Bad Gateway"
Sep 27 10:37:58 atlas go-hass-agent[3218707]: Sep 27 10:37:58.722 ERR Process sensor failed. error="failed to send sensor update request for processes_running: code 502: 502 Bad Gateway"
Sep 27 10:37:58 atlas go-hass-agent[3218707]: Sep 27 10:37:58.737 ERR Process sensor failed. error="failed to send sensor update request for cpufreq_core26_frequency: code 502: 502 Bad Gateway"
Sep 27 10:37:58 atlas go-hass-agent[3218707]: Sep 27 10:37:58.737 ERR Process sensor failed. error="failed to send sensor update request for cpufreq_core52_frequency: code 502: 502 Bad Gateway"
Sep 27 10:37:58 atlas go-hass-agent[3218707]: Sep 27 10:37:58.738 ERR Process sensor failed. error="failed to send sensor update request for core_51_cpu_usage: code 502: 502 Bad Gateway"
Sep 27 10:37:58 atlas go-hass-agent[3218707]: Sep 27 10:37:58.738 ERR Process sensor failed. error="failed to send sensor update request for core_57_cpu_usage: code 502: 502 Bad Gateway"
Sep 27 10:38:01 atlas go-hass-agent[3218707]: Sep 27 10:38:01.510 ERR Process sensor failed. error="failed to send sensor update request for nvme1n1_disk_reads: code 502: 502 Bad Gateway"
Sep 27 10:38:02 atlas go-hass-agent[3218707]: Sep 27 10:38:02.068 ERR Process sensor failed. error="failed to send sensor update request for bytes_sent_throughput: code 502: 502 Bad Gateway"
```

Here is my preferences.toml:

```
version = 'v10.2.1'
registered = true

[mqtt]
server = 'tcp://localhost:1883'
enabled = true

[registration]
server = 'http://10.71.1.15:8123'
token = ...

[hass]
...

[device]
id = '7bb077cc-e359-4f1f-aa14-85b025d80316'
name = 'atlas'
```

Like I said, I am probably missing a step. If someone could take a look and point me in the right direction.

Regards,
Jeff

**Comment by joshuar** [comment]
Hey there, thanks for trying out Go Hass Agent. That's not a great first experience. A `5xx` response usually means something is going wrong on the Home Assistant side. The `502` specifically seems to indicate maybe you have some kind of proxy between the machine running Go Hass Agent and Home Assistant and it is not liking either the the Go Hass Agent requests or the Home Assistant responses?

A couple of things would be worth checking:

- The Home Assistant log. Do you see any error messages that might correlated with the Go Hass Agent requests?
- You could try issuing a request to the URL that the agent is using and viewing the response headers. In the preferences file, take the url in the `apiurl` field and issue a curl request on the command-line like the following:
  ```
  curl -I http://some.host:port/some/path
  ```

  An ideal response would be something like:

  ```
  HTTP/1.1 200 OK
  Referrer-Policy: no-referrer
  X-Content-Type-Options: nosniff
  Server: 
  X-Frame-Options: SAMEORIGIN
  Date: Sat, 28 Sep 2024 23:05:43 GMT
  ```

  The `200 OK` is what you want. If you get something else, it seems like there would be some networking issue (i.e., a proxy problem)?

- Are you using Home Assistant cloud? Do you use the provided cloud endpoint to talk to Home Assistant or do you prefer to use the local address of your server?
